### PR TITLE
bai15

### PR DESCRIPTION
**Pull request by Dinhduy01**
## Related Tickets
- [#Task xxxxx: Content](https://edu-redmine.sun-asterisk.vn/issues/xxxx)
## WHAT
- Bài tập content provider
## Evidence (Screenshot or Video)
3a71b0c2a3557b0b2244
51ee315d22cafa94a3db

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/blob/master/eng/android/coding_convention_android_kotlin.md | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/layout/list_view_item.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        tools:ignore="MissingConstraints">
```
Comment (minhnn-1279) [inline app/src/main/res/layout/list_view_item.xml]: không ignore MissingConstraint nhé :v Nhớ constraint ít nhất 1 đầu theo chiều dọc và 1 đầu theo chiều ngang nhé

File: app/src/main/res/layout/list_view_item.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            android:layout_height="135dp"
+            app:layout_constraintEnd_toEndOf="parent"
+            tools:ignore="MissingConstraints"
+            tools:layout_editor_absoluteY="1dp">
+
+            <TextView
+                android:id="@+id/button_call"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginEnd="40dp"
```
Comment (minhnn-1279) [inline app/src/main/res/layout/list_view_item.xml]: extract dimens

File: app/src/main/java/com/sun/android/bai15_content/CustomAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.BaseAdapter
+import android.widget.TextView
+import android.widget.Toast
+import androidx.core.content.ContextCompat.startActivity
+
+
+class CustomAdapter(context: Context, items: MutableList<Contacts>) : BaseAdapter() {
```
Comment (minhnn-1279) [inline app/src/main/java/com/sun/android/bai15_content/CustomAdapter.kt]: Những bài sau sài RecycleView nhé

Comment (Dinhduy01) [reply app/src/main/java/com/sun/android/bai15_content/CustomAdapter.kt]: yes sir

Comment (ghost) [inline app/src/main/java/com/sun/android/bai15_content/CustomAdapter.kt]: không nên truyền context vào constructor đâu, hàm nào cần thì truyền mà cái adapter này cũng không cần context làm gì

File: app/src/main/java/com/sun/android/bai15_content/CustomAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    var context: Context = context
+    var items: MutableList<Contacts> = items
```
Comment (ghost) [inline app/src/main/java/com/sun/android/bai15_content/CustomAdapter.kt]: cái gì không cần chia sẻ thì để private

File: app/src/main/java/com/sun/android/bai15_content/CustomAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        }
+        return convertView
+    }
+
+    inner class ViewHolder() {
+        var item_name: TextView? = null
+        var item_number: TextView? = null
+        var button_call: TextView? = null
+
+        constructor(view: View) : this() {
```
Comment (ghost) [inline app/src/main/java/com/sun/android/bai15_content/CustomAdapter.kt]: dùng viewBinding

File: app/src/main/java/com/sun/android/bai15_content/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+
+
+class MainActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(com.sun.android.R.layout.activity_main15)
```
Comment (ghost) [inline app/src/main/java/com/sun/android/bai15_content/MainActivity.kt]: dùng viewBinding

File: app/src/main/java/com/sun/android/bai15_content/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+                ActivityCompat.requestPermissions(
+                    this, arrayOf(Manifest.permission.READ_CONTACTS), 1
+                )
+            } else {
+                ActivityCompat.requestPermissions(
+                    this, arrayOf(Manifest.permission.READ_CONTACTS), 1
+                )
+            }
+        }
+        val itemsListView: ListView = findViewById<View>(com.sun.android.R.id.list_view_contacts) as ListView
```
Comment (ghost) [inline app/src/main/java/com/sun/android/bai15_content/MainActivity.kt]: viewBinding

**Pull request merged**